packages/react-utilities/src/hooks/useOnClickOutside.ts

```diff
@@ -96,3 +97,63 @@ const getWindowEvent = (target: Node | Window): Event | undefined => {
 
   return undefined;
 };
+
+const FUI_FRAME_EVENT = 'fuiframefocus';
+
+/**
+ * Since click events do not propagate past iframes, we use focus to detect if a
+ * click has happened inside an iframe, since the only ways of focusing inside an
+ * iframe are:
+ *   - clicking inside
+ *   - tabbing inside
+ *
+ * Polls the value of `document.activeElement`. If it is an iframe, then dispatch
+ * a custom DOM event. When the custom event is received call the provided callback
+ *
+ * @param enableFrameFocusDispatch - boolean flag to start dispatching events
+ * @param targetDocument - the document to dispatch events and set timeouts
+ * @param pollDuration  - in milliseconds
+ */
+const useIFrameFocus = (
+  enableFrameFocusDispatch: boolean,
+  targetDocument: Document | undefined,
+  callback: (e: Event) => void,
+  pollDuration: number = 1000,
+) => {
+  const timeoutRef = React.useRef<number>();
+
+  const listener = React.useCallback(
+    (e: Event) => {
+      if (callback) {
+        callback(e);
+      }
+    },
+    [callback],
+  );
```

Comment:
This requires `callback` to be stable, is it expected? May be use `useEventCallback` instead?

Reply:
well the current case will be some one of `setOpen` callback which is either an event callback already or a `setState` function.... it's hard to say, but I guess it's more idiot proof to use an event callback